File: android/xray/src/main/java/com/applicaster/xray/android/routing/DefaultSinkFilter.java
package com.applicaster.xray.android.routing;

import android.util.Log;

import androidx.annotation.NonNull;

import com.applicaster.xray.routing.ISinkFilter;

public class DefaultSinkFilter implements ISinkFilter {

    private int minLevel = Log.DEBUG;

    public DefaultSinkFilter(int level) {
        minLevel = level;
    }

    @Override
    public boolean accept(@NonNull String loggerName, @NonNull String tag, int level) {
        return minLevel <= level;
    }
}

File: apple/XrayLogger/SinksFilters/DefaultSinkFilter.swift
//
//  DefaultSinkFilter.swift
//  xray
//
//  Created by Anton Kononenko on 7/15/20.
//  Copyright © 2020 Applicaster. All rights reserved.
//

import Foundation

public class DefaultSinkFilter: NSObject, SinkFilterProtocol {
    public private(set) var minLevel: LogLevel = .debug

    public init(level: LogLevel) {
        super.init()
        minLevel = level
    }

    public func accept(loggerSubsystem: String,
                       category: String,
                       logLevel: LogLevel) -> Bool {
        return minLevel.rawValue <= logLevel.rawValue
    }
}
